### PR TITLE
drivers: sensor: apds9960: Update driver to use new driver mechanisms

### PR DESCRIPTION
**Pull request by galak**
Move driver to use gpio_dt_spec and DEVICE_DT_GET for handling access
to bus parent device struct.

Signed-off-by: Kumar Gala <galak@kernel.org>

### CODE REVIEW

**Review by gmarull — CHANGES_REQUESTED** [review]

File: drivers/sensor/apds9960/apds9960.c [hunk 1]
```diff
@@ -360,24 +360,26 @@ static int apds9960_init_interrupt(const struct device *dev)
...
-	drv_data->gpio_pin = config->gpio_pin;
+	if (!device_is_ready(config->int_gpio.port)) {
+		if (config->int_gpio.port) {
+			LOG_ERR("%s: device %s is not ready", dev->name,
+						config->int_gpio.port->name);
+			return -ENODEV;
+		}
+
+		LOG_DBG("%s: int_gpio not defined in DT", dev->name);
+		return 0;
```
[the remark this answers is not in the document]
Comment (galak) [reply drivers/sensor/apds9960/apds9960.c]: Removed the extra check and now just return `-ENODEV`.

**Review by gmarull — DISMISSED** [review]

**Review by gmarull — DISMISSED** [review]

**Comment by galak** [comment]
Fixed `const` usage.

File: drivers/sensor/apds9960/apds9960.c [hunk 2]
```diff
@@ -28,7 +28,7 @@ LOG_MODULE_REGISTER(APDS9960, CONFIG_SENSOR_LOG_LEVEL);
 
 static void apds9960_handle_cb(struct apds9960_data *drv_data)
 {
-	apds9960_setup_int(drv_data, false);
+	apds9960_setup_int(drv_data->dev->config, false);
```
Comment (gmarull) [inline drivers/sensor/apds9960/apds9960.c:31]: note: does this work without de-referencing first to config type? I actually see that drv_data holds a config reference (probably for the work queue context).

Comment (galak) [reply drivers/sensor/apds9960/apds9960.c:31]: I removed the `drv_data` config reference.  The compiler seems fine with it, guessing that there is an implicit cast between the `const void *` and `const struct apds9960_config *`

**Review by MaureenHelm — APPROVED** [review]

**Review by gmarull — APPROVED** [review]

**Pull request merged**